common/scripts/install-run.js

```diff
@@ -0,0 +1,471 @@
+ */
+function _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath) {
+    console.log(`Copying ${sourceNpmrcPath} --> ${targetNpmrcPath}`); // Verbose
+    let npmrcFileLines = fs.readFileSync(sourceNpmrcPath).toString().split('\n');
+    npmrcFileLines = npmrcFileLines.map((line) => (line || '').trim());
+    const resultLines = [];
+    // This finds environment variable tokens that look like "${VAR_NAME}"
+    const expansionRegExp = /\$\{([^\}]+)\}/g;
+    // Comment lines start with "#" or ";"
+    const commentRegExp = /^\s*[#;]/;
+    // Trim out lines that reference environment variables that aren't defined
+    for (const line of npmrcFileLines) {
+        let lineShouldBeTrimmed = false;
+        // Ignore comment lines
+        if (!commentRegExp.test(line)) {
+            const environmentVariables = line.match(expansionRegExp);
+            if (environmentVariables) {
+                for (const token of environmentVariables) {
+                    // Remove the leading "${" and the trailing "}" from the token
+                    const environmentVariableName = token.substring(2, token.length - 1);
+                    // Is the environment variable defined?
+                    if (!process.env[environmentVariableName]) {
+                        // No, so trim this line
+                        lineShouldBeTrimmed = true;
+                        break;
+                    }
+                }
+            }
+        }
+        if (lineShouldBeTrimmed) {
+            // Example output:
+            // "; MISSING ENVIRONMENT VARIABLE: //my-registry.com/npm/:_authToken=${MY_AUTH_TOKEN}"
+            resultLines.push('; MISSING ENVIRONMENT VARIABLE: ' + line);
+        }
+        else {
+            resultLines.push(line);
+        }
+    }
+    fs.writeFileSync(targetNpmrcPath, resultLines.join(os.EOL));
+}
+/**
+ * syncNpmrc() copies the .npmrc file to the target folder, and also trims unusable lines from the .npmrc file.
+ * If the source .npmrc file not exist, then syncNpmrc() will delete an .npmrc that is found in the target folder.
+ *
+ * IMPORTANT: THIS CODE SHOULD BE KEPT UP TO DATE WITH Utilities._syncNpmrc()
+ */
+function _syncNpmrc(sourceNpmrcFolder, targetNpmrcFolder, useNpmrcPublish) {
+    const sourceNpmrcPath = path.join(sourceNpmrcFolder, !useNpmrcPublish ? '.npmrc' : '.npmrc-publish');
+    const targetNpmrcPath = path.join(targetNpmrcFolder, '.npmrc');
+    try {
+        if (fs.existsSync(sourceNpmrcPath)) {
+            _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath);
+        }
+        else if (fs.existsSync(targetNpmrcPath)) {
+            // If the source .npmrc doesn't exist and there is one in the target, delete the one in the target
+            console.log(`Deleting ${targetNpmrcPath}`); // Verbose
+            fs.unlinkSync(targetNpmrcPath);
+        }
+    }
+    catch (e) {
+        throw new Error(`Error syncing .npmrc file: ${e}`);
+    }
+}
+let _npmPath = undefined;
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)